vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/Dashboard.java

```diff
@@ -0,0 +1,1060 @@
+/**
+ * Copyright 2000-2024 Vaadin Ltd.
+ *
+ * This program is available under Vaadin Commercial License and Service Terms.
+ *
+ * See {@literal <https://vaadin.com/commercial-license-and-service-terms>} for the full
+ * license.
+ */
+package com.vaadin.flow.component.dashboard;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.LoggerFactory;
+
+import com.vaadin.flow.component.AttachEvent;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.HasSize;
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.dom.DomEvent;
+import com.vaadin.flow.dom.Element;
+import com.vaadin.flow.internal.JsonSerializer;
+import com.vaadin.flow.shared.Registration;
+
+import elemental.json.JsonArray;
+import elemental.json.JsonObject;
+import elemental.json.JsonType;
+
+/**
+ * @author Vaadin Ltd
+ */
+@Tag("vaadin-dashboard")
+@NpmPackage(value = "@vaadin/polymer-legacy-adapter", version = "24.6.0-alpha2")
+@JsModule("@vaadin/polymer-legacy-adapter/style-modules.js")
+@JsModule("@vaadin/dashboard/src/vaadin-dashboard.js")
+@JsModule("./flow-component-renderer.js")
+@NpmPackage(value = "@vaadin/dashboard", version = "24.6.0-alpha2")
+public class Dashboard extends Component implements HasWidgets, HasSize {
```

Comment:
How would I retrieve the current dashboard config, so that I can persist it somewhere and then later restore it?

In the web component you can just take whatever is in `items`, serialize it to JSON, and then later restore `items` from that JSON. How would I implement something like this in Flow?

Reply:
We have discussed this a lot, and the conclusion was to leave it to the user to choose how to represent their data for serializing/storing. It would require getting the ordered root items through `getChildren` and getting the widgets in sections through those.

Reply:
We might want to let someone DX test this and see what they think. Apart from that I'd suggest to add an example of this to the docs if it isn't already planned.

Reply:
I added a task in the DX tests that require this. I agree that we should add good examples for all these cases to the docs if we keep this API design after the DX tests.